extension-manifest-v3/src/background/utils/request.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { Request } from '@cliqz/adblocker';
+import { parse } from 'tldts-experimental';
+
+export default class ExtendedRequest extends Request {
+  static fromRequestDetails(details) {
```

Comment:
doesn't adblocker/webextension provide similar functionality?

Reply:
The source domain parsed data is lost, so it had to be parsed again. The adblocker-webextension calls `Request` directly, so it can not be overwritten with a version, which keeps the `sourceDomain`.